src/pybind/mgr/orchestrator/module.py

```diff
@@ -844,11 +844,21 @@ def _apply_misc(self, service_type=None, placement=None, unmanaged=False, inbuf=
   ceph orch apply -i <yaml spec>
   ceph orch apply <service_type> <placement> [--unmanaged]
         """
+        results = []
         if inbuf:
             if service_type or placement or unmanaged:
                 raise OrchestratorValidationError(usage)
             content: Iterator = yaml.load_all(inbuf)
-            specs = [ServiceSpec.from_json(s) for s in content]
+
+            specs = []
+            for s in content:
+                if 'service_type' in s and s['service_type'] == 'host':
+                    spec = HostSpec.from_json(s)
+                    spec.service_type = 'host'
```

Comment:
`HostSpec` does not have a `service_type` attribute. One way would be to simply add it. 

I have the feeling that differentiating here between `HostSpec` and  `ServiceSpec` here will need to get refactored eventually.  Maybe we can move this code to `ServiceSpec.from_json()`?